lib/data_collector.go

```diff
@@ -137,9 +137,13 @@ func (dcc *DataCollectorClient) Update(nodeName string, body interface{}) (*http
 	return res, err
 }
 
-func chefAutomateSendMessage(client *DataCollectorClient, nodeName string, body interface{}) error {
-	_, err := client.Update(nodeName, body)
-	return err
+func chefAutomateSendMessage(client *DataCollectorClient, nodeName string, body interface{}) (int, error) {
+	code := 999
+	res, err := client.Update(nodeName, body)
+	if res != nil {
+		code = res.StatusCode
```

Comment:
Why get the status code only when there is an error?

Reply:
Oh it is not only when there is an error, it is **only when there is a response** 😄

Reply:
Sorry, I did not see this. Oh, I see now. The 'res' variable will be 'nil' if there is not an error.